widget/list.go

```diff
@@ -394,16 +398,11 @@ func (lc *listPool) Count() int {
 
 // Obtain returns an item from the pool for use
 func (lc *listPool) Obtain() fyne.CanvasObject {
-	poolLength := len(lc.contents)
-	if poolLength == 0 {
+	if len(lc.contents) == 0 {
 		return nil
 	}
 	item := lc.contents[0]
-	if poolLength > 0 {
-		lc.contents = lc.contents[1:]
-	} else {
-		lc.contents = nil
-	}
+	lc.contents = lc.contents[1:]
 	item.Show()
```

Comment:
I'm not sure it is necessary to Hide/Show pool items since they will not be traversed when the list is rendered. Additionally as Show triggers Refresh and Hide triggers canvas.Refresh, there is a cost here and I have found better performance by instead just calling Refresh after obtaining and updating the item.

Reply:
This should be resolved now.